### ISSUE TITLE
Redundant plotting tools (sorting, extra plots created,..)

### ISSUE THREAD

**Issue by wathen** [fitbenchmarking/fitbenchmarking#242]
**Description of the error**

Plots data, initial guess and best performing minimizer every time there is a better chisq value. 

Sorting function in in  fitting/plotting/plot_helper.py on lines 60 - 65 is of order n^2 and is called each time a new plot is created. This should be moved to the parsers.

**Describe the expected result**
Dramatic speed up

**Describe the actual result**
What was the actual result.

**Comment by wathen** [comment]
Observation:

```
Changing to use argsort in plot_helper.py
mantidpython example_scripts/example_runScripts.py  329.26s user 14.22s system 106% cpu 5:22.58 total

Sorting in parsing
mantidpython example_scripts/example_runScripts.py  332.22s user 14.46s system 106% cpu 5:25.90 total

Master
mantidpython example_scripts/example_runScripts.py  445.65s user 16.82s system 104% cpu 7:20.80 total
```

**Comment by wathen** [comment]
Sorting fixed in #261 and there is no redundant plotting at the moment
